packages/react-aria-components/src/HiddenDateInput.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+
+import {CalendarDate, CalendarDateTime, parseDate, parseDateTime} from '@internationalized/date';
+import {DateFieldState, DatePickerState, DateSegmentType} from 'react-stately';
+import React, {useEffect} from 'react';
+import {useVisuallyHidden} from 'react-aria';
+
+interface AriaHiddenDateInputProps {
+  /**
+   * Describes the type of autocomplete functionality the input should provide if any. See [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefautocomplete).
+   */
+  autoComplete?: string,
+  /** HTML form input name. */
+  name?: string,
+  /** Sets the disabled state of the input. */
+  isDisabled?: boolean
+}
+
+interface HiddenDateInputProps extends AriaHiddenDateInputProps {
+  /**
+   * State for the input.
+   */
+  state: DateFieldState | DatePickerState
+}
+
+export interface HiddenDateAria {
+  /** Props for the container element. */
+  containerProps: React.HTMLAttributes<HTMLDivElement>,
+  /** Props for the hidden input element. */
+  inputProps: React.InputHTMLAttributes<HTMLInputElement>
+}
+
+export function useHiddenDateInput(props: HiddenDateInputProps, state: DateFieldState | DatePickerState) {
+  let {
+    autoComplete,
+    isDisabled,
+    name
+  } = props;
+  let [dateValue, setDateValue] = React.useState('');
+  let {visuallyHiddenProps} = useVisuallyHidden();
+
+  let inputStep = 60;
+  if (state.granularity === 'second') {
+    inputStep = 1;
+  } else if (state.granularity === 'hour') {
+    inputStep = 3600; 
+  }
+
+  useEffect(() => {
+    if (state.value == null) {
+      setDateValue('');
+    } else {
+      setDateValue(state.value.toString());
+    }
+  }, [state.value]);
```

Comment:
any way we can derive the state in render instead of causing a second render anytime `state.value` changes?